python_modules/dagster/dagster/core/decorators.py

```diff
@@ -1,17 +1,29 @@
 import inspect
+from collections import namedtuple
 from functools import wraps
 from dagster import check
 from .definitions import (
     SolidDefinition,
     InputDefinition,
     OutputDefinition,
     DagsterInvalidDefinitionError,
+    Result,
+    check_argument_def_dict,
 )
 
 # Error messages are long
 # pylint: disable=C0301
 
 
+class MultipleResults(namedtuple('_MultipleResults', 'results')):
+    def __new__(cls, *results):
+        return super(MultipleResults, cls).__new__(
+            cls,
+            # XXX(freiksenet): should check.list_param accept tuples from rest?
```

Comment:
If we want a version that can take an array or tuple we can just add a static helper method